packages/engine/src/lib/handler/piece-executor.ts

```diff
@@ -38,28 +39,31 @@ const executeAction: ActionHandler<PieceAction> = async ({ action, executionStat
         status: StepOutputStatus.RUNNING,
     })
 
-    try {
-        assertNotNullOrUndefined(action.settings.actionName, 'actionName')
+    const { data: executionStateResult, error: executionStateError } = await tryCatchAndThrowEngineError((async () => {
+        if (isNil(action.settings.actionName)) {
+            throw new EngineGenericError('ActionNameNotSetError', 'Action name is not set')
+        }
+        
         const { pieceAction, piece } = await pieceLoader.getPieceAndActionOrThrow({
             pieceName: action.settings.pieceName,
             pieceVersion: action.settings.pieceVersion,
             actionName: action.settings.actionName,
             pieceSource: constants.piecesSource,
         })
-
+    
         const { resolvedInput, censoredInput } = await constants.propsResolver.resolve<StaticPropsValue<PiecePropertyMap>>({
             unresolvedInput: action.settings.input,
             executionState,
         })
-
+    
         stepOutput.input = censoredInput
-
+    
         const { processedInput, errors } = await propsProcessor.applyProcessorsAndValidators(resolvedInput, pieceAction.props, piece.auth, pieceAction.requireAuth, action.settings.propertySettings)
         if (Object.keys(errors).length > 0) {
-            throw new Error(JSON.stringify(errors, null, 2))
+            throw new EngineGenericError('PieceActionExecutionError', `Piece action execution error: ${JSON.stringify(errors, null, 2)}`)
```

Comment:
This is wrong, that happens when the user enters invalid validation, we watch this error in the piece and consider it user error.

And we depend on the string be json because we display this inside the piece step.